plugins/out_stackdriver/stackdriver_http_request.c

```diff
@@ -0,0 +1,453 @@
+                          flb_sds_len(http_request->userAgent));
+
+    msgpack_pack_str(mp_pck, 8);
+    msgpack_pack_str_body(mp_pck, "remoteIp", 8);
+    msgpack_pack_str(mp_pck, flb_sds_len(http_request->remoteIp));
+    msgpack_pack_str_body(mp_pck, http_request->remoteIp, 
+                          flb_sds_len(http_request->remoteIp));
+
+    msgpack_pack_str(mp_pck, 8);
+    msgpack_pack_str_body(mp_pck, "serverIp", 8);
+    msgpack_pack_str(mp_pck, flb_sds_len(http_request->serverIp));
+    msgpack_pack_str_body(mp_pck, http_request->serverIp, 
+                          flb_sds_len(http_request->serverIp));
+
+    msgpack_pack_str(mp_pck, 7);
+    msgpack_pack_str_body(mp_pck, "referer", 7);
+    msgpack_pack_str(mp_pck, flb_sds_len(http_request->referer));
+    msgpack_pack_str_body(mp_pck, http_request->referer, 
+                          flb_sds_len(http_request->referer));
+
+    msgpack_pack_str(mp_pck, 8);
+    msgpack_pack_str_body(mp_pck, "protocol", 8);
+    msgpack_pack_str(mp_pck, flb_sds_len(http_request->protocol));
+    msgpack_pack_str_body(mp_pck, http_request->protocol, 
+                          flb_sds_len(http_request->protocol));
+
+    /* Integer sub-fields */
+    msgpack_pack_str(mp_pck, 11);
+    msgpack_pack_str_body(mp_pck, "requestSize", 11);
+    msgpack_pack_int64(mp_pck, http_request->requestSize);
+
+    msgpack_pack_str(mp_pck, 12);
+    msgpack_pack_str_body(mp_pck, "responseSize", 12);
+    msgpack_pack_int64(mp_pck, http_request->responseSize);
+
+    msgpack_pack_str(mp_pck, 6);
+    msgpack_pack_str_body(mp_pck, "status", 6);
+    msgpack_pack_int64(mp_pck, http_request->status);
+
+    msgpack_pack_str(mp_pck, 14);
+    msgpack_pack_str_body(mp_pck, "cacheFillBytes", 14);
+    msgpack_pack_int64(mp_pck, http_request->cacheFillBytes);
+
+    /* Boolean sub-fields */
+    msgpack_pack_str(mp_pck, 11);
+    msgpack_pack_str_body(mp_pck, "cacheLookup", 11);
+    if (http_request->cacheLookup == FLB_TRUE) {
+        msgpack_pack_true(mp_pck);
+    }
+    else {
+        msgpack_pack_false(mp_pck);
+    }
+
+    msgpack_pack_str(mp_pck, 8);
+    msgpack_pack_str_body(mp_pck, "cacheHit", 8);
+    if (http_request->cacheLookup == FLB_TRUE) {
+        msgpack_pack_true(mp_pck);
+    }
+    else {
+        msgpack_pack_false(mp_pck);
+    }
+
+    msgpack_pack_str(mp_pck, 30);
+    msgpack_pack_str_body(mp_pck, "cacheValidatedWithOriginServer", 30);
+    if (http_request->cacheValidatedWithOriginServer == FLB_TRUE) {
+        msgpack_pack_true(mp_pck);
+    }
+    else {
+        msgpack_pack_false(mp_pck);
+    }
+}
+
+/* latency should be in the format: 
+ *      whitespace (opt.) + integer + point & decimal (opt.)
+ *      + whitespace (opt.) + "s" + whitespace (opt.) 
+ */
+static void validate_latency(msgpack_object_str latency_in_payload, 
+                             struct http_request_field *http_request) {
+    flb_sds_t pattern = flb_sds_create("^\\s*\\d\\d*(.\\d\\d*)?\\s*s\\s*$");
```

Comment:
can you do `\\d+` instead of `\\d\\d*`  in fluentbit?

Reply:
Done